src/Classical/linear_code.jl

```diff
@@ -396,6 +396,19 @@ end
 #############################
      # general functions
 #############################
+function information_set(C::AbstractLinearCode)
+    """
+    Returns the indexs of the pivot columns and an information set which 
+    is the submatrix of G given by these columns.
+    """
+    if C.G == C.G_stand
```

Comment:
This is almost never going to happen. I'd suggest dropping this case. Also, I think it would go to `C.k` and not `C.n`.

But now we have to jump into something deeper that you probably haven't gotten to you. We probably haven't even fixed this completely throughout the library yet either. Some code families don't have a `C.G` or `C.H` and they are generated only as needed. Therefore, in functions which work on arbitrary linear codes, we should use `G = generator_matrix(C)`.  This will be equivalent to what you wrote in most cases and then generated as needed in the others.

Reply:
Ill check if doing this with lazy initialization via getproperty is feasible and won't require big changes in other parts of the code.